ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -48,33 +49,63 @@ import kotlin.collections.component2
 import kotlin.collections.set
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.reactive.asFlow
 
 /** View model for the Data Collection fragment. */
+@OptIn(ExperimentalCoroutinesApi::class)
 @HiltViewModel
 class DataCollectionViewModel
 @Inject
 internal constructor(
   private val viewModelFactory: ViewModelFactory,
   private val submissionRepository: SubmissionRepository,
+  private val locationOfInterestRepository: LocationOfInterestRepository,
   private val locationOfInterestHelper: LocationOfInterestHelper,
   private val popups: Provider<EphemeralPopups>,
   private val navigator: Navigator,
   @ApplicationScope private val externalScope: CoroutineScope,
   @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
-  private val savedStateHandle: SavedStateHandle
+  private val savedStateHandle: SavedStateHandle,
+  surveyRepository: SurveyRepository,
 ) : AbstractViewModel() {
 
-  val submission: @Hot(replays = true) LiveData<Submission>
-  val jobName: @Hot(replays = true) LiveData<String>
-  val loiName: @Hot(replays = true) LiveData<String>
+  // TODO(#1541): Set loiId once Suggest LOI task is completed.
+  private val loiId: StateFlow<String?> = MutableStateFlow(savedStateHandle["locationOfInterestId"])
+  private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
+  private val job: Job =
+    activeSurvey.getJob(requireNotNull(savedStateHandle["jobId"])).orElseThrow()
+
+  val surveyId: String = surveyRepository.lastActiveSurveyId
```

Comment:
Should we pass the surveyId as well similar to loiId and jobId?

Reply:
I don't have a strong opinion here. The only issue I had is that we are coupling the active survey with data collection fragment. In theory, DataCollectionFragment shouldn't worry about what's currently loaded or not. I'll defer to your judgement here. Thanks!

Reply:
I think since we need to get the activeSurvey from the SurveyRepository anyway I'll stick with getting the survey id from the SurveyRepository,